tests/test_unittest/test_gtscript_frontend.py

```diff
@@ -135,6 +135,46 @@ def definition_func(inout_field: gtscript.Field[float]):
                 definition_func, "test_missing_nested_symbol", module, externals=externals
             )
 
+    def test_undecorated_delay(self, id_version):
+        A = 0
+
+        def undecorated_function():
+            return A
+
+        module = f"TestInlinedExternals_test_undecorated_delay_{id_version}"
+        externals = {"func": undecorated_function}
+
+        A = 1
+
+        # Direct function
+        def definition_func(inout_field: gtscript.Field[float]):
+            from gt4py.__gtscript__ import PARALLEL, computation, interval
+
+            with computation(PARALLEL), interval(...):
+                inout_field = undecorated_function()
+
+        stencil_id, def_ir = compile_definition(
+            definition_func, "test_undecorated_delay", module, externals=externals
+        )
+
+        stmt = def_ir.computations[0].body.stmts[0]
+        assert isinstance(stmt.value, gt_ir.ScalarLiteral) and stmt.value.value == 1
+
+        # As external
+        def definition_func(inout_field: gtscript.Field[float]):
+            from gt4py.__externals__ import func
+            from gt4py.__gtscript__ import PARALLEL, computation, interval
+
+            with computation(PARALLEL), interval(...):
+                inout_field = func()
+
+        stencil_id, def_ir = compile_definition(
+            definition_func, "test_undecorated_delay", module, externals=externals
+        )
+
+        stmt = def_ir.computations[0].body.stmts[0]
+        assert isinstance(stmt.value, gt_ir.ScalarLiteral) and stmt.value.value == 1
+
```

Comment:
I think a test is missing here to check that the decorated function will freeze the value of A even if it is changed later.

Reply:
Once this new test is added, I will approve the PR

Reply:
I'll fix that tonight. Thanks for reminding me.